external/urlfrontier/src/main/java/com/digitalpebble/stormcrawler/urlfrontier/StatusUpdaterBolt.java

```diff
@@ -170,20 +171,28 @@ public void prepare(
         }
 
         if (address == null) {
-            channel =
-                    ManagedChannelUtil.createChannel(
-                            ConfUtils.getString(
-                                    stormConf, URLFRONTIER_HOST_KEY, URLFRONTIER_DEFAULT_HOST),
-                            ConfUtils.getInt(
-                                    stormConf, URLFRONTIER_PORT_KEY, URLFRONTIER_DEFAULT_PORT));
-        } else {
-            channel = ManagedChannelUtil.createChannel(address);
+            String host =
+                    ConfUtils.getString(stormConf, URLFRONTIER_HOST_KEY, URLFRONTIER_DEFAULT_HOST);
+            int port = ConfUtils.getInt(stormConf, URLFRONTIER_PORT_KEY, URLFRONTIER_DEFAULT_PORT);
+            address = host + ":" + port;
         }
 
+        channel = ManagedChannelUtil.createChannel(address);
+        channel.notifyWhenStateChanged(ConnectivityState.SHUTDOWN, () -> onChannelStateChange(ConnectivityState.SHUTDOWN));
         URLFrontierStub frontier = URLFrontierGrpc.newStub(channel).withWaitForReady();
         requestObserver = frontier.putURLs(this);
     }
 
+    private void onChannelStateChange(ConnectivityState state) {
+        ConnectivityState newState = channel.getState(true);
+        LOG.debug("Channel state changed from {} to {}", state, newState);
+        if (state == ConnectivityState.TRANSIENT_FAILURE) {
```

Comment:
equals() ?

Reply:
== seems to be equally good ([StackOverflow](https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals)). Nevertheless, I can change it if wished